app/controllers/assignments_controller.rb

```diff
@@ -62,7 +62,7 @@ def show
   def peer_review
     assignment = Assignment.find(params[:id])
     @assignment = assignment.is_peer_review? ? assignment : assignment.pr_assignment
-    if @assignment.nil? || @assignment.is_hidden
+    unless @assignment.nil? || !allowed_to?(:see_hidden?)
```

Comment:
Please see my previous comment, the logic is still backwards here. It should be either:

```ruby
if @assignment.nil? || !allowed_to(:see_hidden?)

# OR 

unless !@assignment.nil? || allowed_to(:see_hidden?)
```